### ISSUE TITLE
Regression in v2026.01.12: Bindings of unquoted gensyms not detected anymore

### ISSUE THREAD

**Issue by bitti** [clj-kondo/clj-kondo#2747]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2026.01.12

**macro usage**

Not in macro usage, but in macro definition.

**platform**

Several

**problem**

The bindings of unquoted [gensyms](https://clojure.org/guides/weird_characters#gensym), as they can appear in nested syntax quotes, are not detected anymore.

**repro**

Linting

```clojure
(defmacro def-some-macro
  []
  `(defmacro ~'some-macro
     [x#]
     `(list ~x#)))
```

Yields: `error: Unresolved symbol: x#` 

**expected behavior**

No error should be generated if there is a binding.

**Comment by borkdude** [comment]
Thanks
